### PR TITLE
feat: create ProgressBar component

### PR DESCRIPTION
**Pull request by nartydev**
image

### CODE REVIEW

File: src/components/ProgressBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export const ProgressBar: FC<IProps> = ({ percentage }) => {
+  return (
+    <div className={s['container']}>
+      <Typography>{percentage}%</Typography>
+      <div className={s['progress-bar']}>
+        <div
+          className={s['progress-bar-active']}
+          style={{ width: `${percentage}%` }}
+        />
```
Comment (moreirathomas) [inline src/components/ProgressBar/index.tsx:15]: Suggestion:
Use native and semantic [`<progress>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress) element and only tweak its style.

```suggestion
        <progress
          className={s['progress-bar-custom-styles-here']}
          max={props.max}
          value={props.value}
        />
```

File: src/components/ProgressBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { FC } from 'react'
+
+import { Typography } from '../Typography'
+import s from './core/progressbar.module.scss'
+import { IProps } from './core/progressbar.typings'
+
+export const ProgressBar: FC<IProps> = ({ percentage }) => {
+  return (
+    <div className={s['container']}>
+      <Typography>{percentage}%</Typography>
```
Comment (moreirathomas) [inline src/components/ProgressBar/index.tsx:10]: If we use the `<progress>` element, the variables in scope would be `max` and `value`. 

Passing the percentage would be overkill as all the info needed to compute it here is available.

```suggestion
      <Typography>{(100 * props.value) / props.max }%</Typography>
```

Or extracted as a function

```ts
const percentage = (value: number, max: number) => (100 * value) / max
```

File: src/components/ProgressBar/core/progressbar.typings.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface IProps {
+  percentage: number
+}
```
Comment (moreirathomas) [inline src/components/ProgressBar/core/progressbar.typings.tsx:3]: Follow up on the suggestion:

```suggestion
export interface IProps {
  value: number
  max: number
}
```

**Review by moreirathomas — CHANGES_REQUESTED** [review]
Nice ! I left suggestion to use a more semantic html element.

Also, I'm in favor of using the newly created `ProgressBar` inside `@/pages/Run/components/RunResultDisplay.tsx`.

This would allow to increasingly integrate the new component into the app and see how things go step by steps instead of facing a _bing bang_ (and thus dangerous) final integration :)

```diff
const ProgressSection: React.FC<RunProgress & { stop: () => false | void }> = ({
  doneCount,
  maxCount,
  elapsed,
  timeout,
}) => (
  <section>
    <Typography element="h3">Recording Progress</Typography>
    <Button
      text="Stop run"
      onClick={() => stop()}
      style="outlined"
      iconEnd={StopCircle}
    />
    <p>
-     {doneCount} / {maxCount} requests ({(100 * doneCount) / maxCount}%)
      <br />
-       <progress max={maxCount} value={doneCount} />
+       <ProgressBar max={maxCount} value={doneCount} />
      <br />
      {((timeout - elapsed) / 1_000_000_000).toFixed(0)}s
    </p>
  </section>
)
```

By the way this also begs the question: should `ProgressBar` also display the the absolute values too, like the previous skeleton did?
`15 / 200 requests (7.5%)` instead of `7.5%` ?

File: src/components/ProgressBar/index.tsx (hunk 1 above)
Comment (nartydev) [reply src/components/ProgressBar/index.tsx:15]: For custom styles of progress bar with semantic element `progress`, Firefox doesn't support this but we can try like this https://caniuse.com/?search=progress%20webkit :)

**Pull request merged**